bridgenode/flatfileworker.go

```diff
@@ -87,8 +86,21 @@ func flatFileWorker(
 		panic(err)
 	}
 
-	// for the undofiles
+	ud := <-proofChan
```

Comment:
We need a `for{}` loop here as the program will execute this function exactly once and then exit.

```
for {                                               
                ud := <-proofChan                           
                err = ff.writeProofBlock(ud)             
                if err != nil {
                        panic(err)                       
                }                                                                                                       
        }
```

Something like the code above.